src/text.c

```diff
@@ -298,9 +299,10 @@ LedStateMaskText(struct xkb_context *ctx, enum xkb_state_component mask)
 
         mask &= ~(1u << i);
 
+        assert(LookupValue(lookup, 1u << i) != NULL);
```

Comment:
Is the assert safe? Is there any way we can get an unknown value from user input here (maybe xkbcommon-x11)?

Reply:
> maybe xkbcommon-x11

That would be it, because this does not use the same lookup table.

Well, there is another one: the field can be legitimately `0`, so the caller has to ensure to use the `*MaskText` functions with a non-`0` value present in the lookup table.

It would make sense to change the definition of `LookupEntry` like so:

```diff
typedef struct {
     const char *name;
-    uint32_t value;
+    uint32_t xkbcommon;
+    uint32_t x11;
} LookupEntry;
```

then use the *same lookup table* in both our parser and in the X11 API. Then we can have a check for `NULL` in the dump function, so that we exit with a useful error and never print garbage. What do you think?